src/main/java/com/jenkinsci/plugins/badge/action/AbstractAction.java

```diff
@@ -42,4 +45,79 @@ public void setId(String id) {
   public String getId() {
     return id;
   }
+
+  /**
+   * Is the icon reference a Jenkins symbol name.
+   * @param icon icon reference
+   * @return {@code true} if the icon reference is a Jenkins symbol name
+   */
+  static boolean isJenkinsSymbolRef(@Nullable String icon) {
+    return icon != null && icon.startsWith("symbol-");
+  }
+
+  /**
+   * If the given icon reference is a Jenkins symbol append the ionicons plugin class to the icon reference
+   * if not already included.
+   * @param icon icon reference
+   * @return icon reference with ionicons plugin class
+   */
+  @CheckForNull
+  static String getJenkinsSymbolIconPath(@Nullable String icon) {
+    if (isJenkinsSymbolRef(icon) && !icon.contains("plugin-ionicons-api")) {
```

Comment:
Wondering why we should limit this feature to the ionicons plugin alone and not also enable https://github.com/jenkinsci/font-awesome-api-plugin or https://github.com/jenkinsci/custom-folder-icon-plugin/blob/main/src/main/java/jenkins/plugins/foldericon/Emojis.java (and potentially others).

Reply:
Thanks for the comment @strangelookingnerd !  That's a very interesting suggestion, though I think it is outside of the scope of this pull request.  This pull request is already making a significant improvement to the plugin.

The plugin does not have a current maintainer.  It needs to be adopted so that a current maintainer can review the pull request, test it, merge it, and release it.  I've been testing this pull request because I'm an interested user of the plugin and I maintain the embeddable build status plugin.

Are you interested in becoming a [maintainer of the plugin](https://www.jenkins.io/doc/developer/plugin-governance/adopt-a-plugin/)?

Are you interested in submitting the pull request to add the additional support?

I've been submitting changes to the pull request while I've been testing it.  If you adopt the plugin, you'll be able to do the same and help @cronik on the journey to the release of this pull request.

Reply:
This is very good idea, open new issue (enhancement) and I will try to implement it here. Thx